### ISSUE TITLE
How to choose the best model from the saved checkpoint?

### ISSUE THREAD

**Issue by zeal-github** [THU-DA-6D-Pose-Group/GDR-Net#79]
Hi, thanks for releasing your code, that helps me a lot! 
Have you found the results of the last several saved checkpoints are different and the best result on the test dataset is always not the model_final.pth? 
Do you make choices from the last several saved checkpoint, or just choose the result of the last model(model_final.pth) to report?

**Comment by lu-ming-lei** [comment]
hi, what is the specific information of your experiment? 
In my experiments, I found that epoch needs to be added. According to the epoch given by the author, it often does not reach the optimal convergence.

**Comment by zeal-github** [comment]
@lu-ming-lei Hi, I follow the parameters provided by the author. I did find that the experiment's result is slightly lower than the paper(although very close). So I think maybe the reported results are not the last model.
Do you mean that by increasing the epochs, the result will be better? How much epoch did you add and how much benefit did you get?

**Comment by lu-ming-lei** [comment]
I also found the same problem, which is slightly lower than the value given in the article. 
I mean, the results near  model_final.pth  will fluctuate, such as add (s)

**Comment by wangg12** [comment]
> Hi, thanks for releasing your code, that helps me a lot! Have you found the results of the last several saved checkpoints are different and the best result on the test dataset is always not the model_final.pth? Do you make choices from the last several saved checkpoint, or just choose the result of the last model(model_final.pth) to report?

We always used the last checkpoint since we do not have a validation set. But you can use some validation set (for example, split some samples from the original training set for validation) to help you choose a best model.
